src/wp-includes/rest-api/class-wp-rest-server.php

```diff
@@ -1032,9 +1032,9 @@ public function dispatch( $request ) {
 				 *
 				 * @since 4.7.0
 				 *
-				 * @param WP_HTTP_Response|WP_Error $response Result to send to the client. Usually a WP_REST_Response or WP_Error.
-				 * @param array                     $handler  Route handler used for the request.
-				 * @param WP_REST_Request           $request  Request used to generate the response.
+				 * @param WP_REST_Response|WP_HTTP_Response|WP_Error $response Result to send to the client. Usually a WP_REST_Response or WP_Error.
```

Comment:
While we're here, we should probably add `mixed` as a type too. Routes can just return the data they want to be rendered, and `rest_ensure_response` will make it a response object.

Reply:
Added in https://github.com/WordPress/wordpress-develop/pull/166/commits/ecc98b8d29c5c0c11c19183f0d78adc7b3f3ad4f.